activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -1003,28 +1003,33 @@ def self.discard_unowned_pools(pid_map) # :nodoc:
         end
       end
 
-      attr_reader :prevent_writes
-
       def initialize
         # These caches are keyed by spec.name (ConnectionSpecification#name).
         @owner_to_pool = ConnectionHandler.create_owner_to_pool
-        @prevent_writes = false
 
         # Backup finalizer: if the forked child never needed a pool, the above
         # early discard has not occurred
         ObjectSpace.define_finalizer self, ConnectionHandler.unowned_pool_finalizer(@owner_to_pool)
       end
 
+      def prevent_writes # :nodoc:
+        Thread.current[:prevent_writes]
+      end
+
+      def prevent_writes=(prevent_writes) # :nodoc:
+        Thread.current[:prevent_writes] = prevent_writes
```

Comment:
It feels a little off that if we had multiple `ConnectionHandlers` this assignment would apply to all of them

Reply:
That makes sense. It feels off, but I don't see an actual problem caused by it.